app/src/main/java/com/tugraz/quizlet/backend/database/DBManager.kt

```diff
@@ -1,54 +1,94 @@
+import io.realm.kotlin.where
 import io.realm.mongodb.App
 import io.realm.mongodb.AppException
 import io.realm.mongodb.Credentials
 import io.realm.mongodb.mongo.MongoClient
 import io.realm.mongodb.mongo.MongoCollection
 import io.realm.mongodb.mongo.MongoDatabase
 import io.realm.mongodb.sync.SyncConfiguration
+import kotlinx.coroutines.awaitAll
 import org.bson.Document
 import org.bson.types.ObjectId
 import java.util.logging.Logger
 import kotlin.jvm.Throws
-import kotlin.math.absoluteValue
+import com.tugraz.quizlet.backend.database.model.User as User
 
 class DBManager(private val quizletApp: App) : DBInterface {
     companion object {
         val LOG: Logger = Logger.getLogger(DBManager::class.java.name)
     }
 
+    private var anon: io.realm.mongodb.User? = null
+
     private var user: io.realm.mongodb.User? = null
     private var realm: Realm? = null
 
     init {
+        //addUser("emilia0@test.com", "itsmeemilia")
+
+        //Thread.sleep(1000)
+
+        //loginUser("something@something.com", "hihihizuujkj")
+
+        //LOG.fine("logged in user " + quizletApp.currentUser())
+        /*
+        var category = Question_category("description", "category name")
+        var listWrong: RealmList<String> = RealmList()
+        listWrong.add("wrong answer 1")
+        listWrong.add("wrong answer 2")
+        listWrong.add("wrong answer 3")
+
+        var question = Question(ObjectId(), category, "This is my question", "this is the right answer")*/
+
+        //loginUser("emilia0@test.com", "itsmeemilia")
+
+        //Thread.sleep(100)
 
+        /*
+        val questiongen = Generator8()
+        questiongen.foreach { q: Question -> addQuestion(q) }
+
+        */
+
+        /** This needs to be here for the public question **/
+        anonymousLogin()
+
+
+        loginUser("emilia0@test.com", "itsmeemilia")
+        updateUserHighscore(5)
     }
 
-    override fun addQuestion(question: Question) {
-        user = quizletApp.currentUser()
-        question.userCreated = user!!.id;
-        val config = SyncConfiguration.Builder(user!!, user!!.id)
+    fun anonymousLogin() {
+        val anonymousCredentials: Credentials = Credentials.anonymous()
+        quizletApp.login(anonymousCredentials)
+        anon = quizletApp.currentUser()
+    }
+
+    override fun addQuestion(question: Question){
+        //user = quizletApp.currentUser()
+        question.userCreated = anon!!.id
+        val config = SyncConfiguration.Builder(anon!!, anon!!.id)
             .allowWritesOnUiThread(true)
             .build()
 
         realm = Realm.getInstance(config)
         realm?.executeTransactionAsync() { transactionRealm ->
             transactionRealm.insert(question)
-            LOG.severe("did something")
         }
     }
 
     override fun getAllQuestions(): ImmutableList<Question> {
-        user = quizletApp.currentUser()
-        val config = SyncConfiguration.Builder(user!!, user!!.id)
+        //user = quizletApp.currentUser()
```

Comment:
add a check if user is null before instead of using !!

Reply:
should never be null as questions can only be queried if the user is logged in in frontend